tests/all/test_values.rs

```diff
@@ -524,83 +519,6 @@ fn test_metadata() {
     // assert!(!vec_val.has_metadata());
     // assert!(!fn_val.has_metadata());
 
-    bool_val.set_metadata(md_string, 3);
```

Comment:
If you get rid of this code then we don't test get/set metadata at all. Could you update it to test a couple of instructions and non-instructions, please?

Reply:
You're totally right. In doing that I also spotted that it could never work since set_metadata et all were not implemented for InstructionValue. So I've moved it all over to InstructionValue and added some tests.